### PR TITLE
fix reference link

### PR DESCRIPTION
**Pull request by coliu19**
The linter's doc link miss a -"ruleset", Here I fixed this missing part.

### CODE REVIEW

File: api/internal/data/rules-guidelines.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
     "analyzer_name_id": "guidelines",
     "title": "POST operations which create objects return 201 Created, with a full or reference-only representation.",
     "description": "POST operations which create objects return 201 Created, with a full or reference-only representation.",
-    "mitigation": "Please update responses for the items identified. [Reference](https://developer.cisco.com/docs/api-insights/#!rest-guidelines)",
+    "mitigation": "Please update responses for the items identified. [Reference](https://developer.cisco.com/docs/api-insights/#!rest-guidelines-ruleset)",
```
Comment (npateriya) [inline api/internal/data/rules-guidelines.json:95]: We want to point to a specific rule for example https://developer.cisco.com/docs/api-insights/#!rest-style/post not just top level page. 

Please refer https://developer.cisco.com/docs/api-insights/#!rest-guidelines-ruleset/rest-style-rules individual rule "Reference" for the exact location example 

This applies to all refrences.

**Review by npateriya — CHANGES_REQUESTED** [review]
We want to point to a specific rule for example https://developer.cisco.com/docs/api-insights/#!rest-style/post not just top level page https://developer.cisco.com/docs/api-insights/#!rest-guidelines-ruleset.

Please refer https://developer.cisco.com/docs/api-insights/#!rest-guidelines-ruleset/rest-style-rules individual rule "Reference" for the exact location example

This applies to all refrences in rule-guidelines.json.

File: api/internal/data/rules-guidelines.json (hunk 1 above)
Comment (coliu19) [reply api/internal/data/rules-guidelines.json]: Updated most of them. There 15 rules that miss descriptions in all the docs. They are most about version and security. Examples are like oas2-path-based-versioning-major-only, oas2-https-only

File: api/internal/data/rules-completeness.json [hunk 2]
```diff
@@ -4,37 +4,37 @@
...
+    "mitigation": "Please fix the items detected to adhere to the OpenAPI v3 specifications. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/oas23-schema-well-formed-openapi-document)",
     "severity": "error"
   },
   {
     "name_id": "oas3-missing-schema-definition",
     "analyzer_name_id": "completeness",
     "title": "There is no schema attribute for a component.",
     "description": "Some schema definitions are missing.",
-    "mitigation": "Please add schema definitions for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness/oas23-missing-schema-definition-missing-schema-definition)",
+    "mitigation": "Please add schema definitions for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/oas23-missing-schema-definition-missing-schema-definition)",
```
Comment (xinjixi) [inline api/internal/data/rules-completeness.json:23]: This link doesn't navigate to specific rule, https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/oas23-missing-schema-definition-missing-schema-definition

File: api/internal/data/rules-completeness.json [hunk 3]
```diff
@@ -4,37 +4,37 @@
...
     "severity": "error"
   },
   {
     "name_id": "oas2-missing-schema-definition",
     "analyzer_name_id": "completeness",
     "title": "There is no schema attribute for a component.",
     "description": "Some schema definitions are missing.",
-    "mitigation": "Please add schema definitions for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness/oas23-missing-schema-definition-missing-schema-definition)",
+    "mitigation": "Please add schema definitions for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/oas23-missing-schema-definition-missing-schema-definition)",
```
Comment (xinjixi) [inline api/internal/data/rules-completeness.json:31]: This link doesn't navigate to specific rule as above

File: api/internal/data/rules-completeness.json [hunk 4]
```diff
@@ -4,37 +4,37 @@
...
     "severity": "error"
   },
   {
     "name_id": "general-schema-definition",
     "analyzer_name_id": "completeness",
-    "title": "Some of the defined schema use object as a final field when describing their object structure. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness/general-schema-definatio-generic-schema-definition)",
+    "title": "Some of the defined schema use object as a final field when describing their object structure. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/general-schema-definatio-generic-schema-definition)",
```
Comment (xinjixi) [inline api/internal/data/rules-completeness.json:37]: This link doesn't navigate to specific rule as above.

File: api/internal/data/rules-completeness.json [hunk 5]
```diff
@@ -44,95 +44,95 @@
     "analyzer_name_id": "completeness",
     "title": "2XX (except 204) and 4xx responses must have a response schema defined.",
     "description": "Some responses do not define a schema.",
-    "mitigation": "Please add a schema for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness/oas2-3-missing-returned-representation-missing-returned-representation)",
+    "mitigation": "Please add a schema for the items detected. [Reference](https://developer.cisco.com/docs/api-insights/#!documentation-completeness-ruleset/oas2-3-missing-returned-representation-missing-returned-representation)",
```
Comment (xinjixi) [inline api/internal/data/rules-completeness.json:47]: This link doesn't navigate to specific rule as above.

**Review by xinjixi — COMMENTED** [review]
Hi @coliu19 , I checked the links, some links don't navigate to specific rule even when the anchor is given, should we create a PR to fix anchors in docs?

File: api/internal/data/rules-guidelines.json (hunk 1 above)
Comment (xinjixi) [reply api/internal/data/rules-guidelines.json]: Maybe we could add them in additional rules section of the docs and update the links as https://developer.cisco.com/docs/api-insights/#!rest-guidelines-ruleset/rest-additional-rules

File: api/internal/data/rules-completeness.json (hunk 2 above)
Comment (coliu19) [reply api/internal/data/rules-completeness.json:23]: Yes, should have further fix there.

**Pull request merged**